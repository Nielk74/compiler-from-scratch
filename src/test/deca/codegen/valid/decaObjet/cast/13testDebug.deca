// Ok

{
  Object o = null;
  
  if (o == null) {
    println("ok 1");
  } else {
    println("ko 1");
  }

  o = (Object)(o);

  if (o == null) {
    println("ok 2");
  } else {
    println("ko 2");
  }
   
  o = (Object)(null);

  if (o == null) {
    println("ok 3");
  } else {
    println("ko 3");
  }

  if (o instanceof Object) {
    println("ko 4 - because o is null");
  } else {
    println("ok 4");
  }
}
