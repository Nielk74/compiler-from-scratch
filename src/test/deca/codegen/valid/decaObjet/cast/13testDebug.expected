ok 1
ok 2
ok 3
ko 4 - because o is null
